src/comm/TcpAcceptor.cc

```diff
@@ -390,10 +390,17 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     details->local = *gai;
     Ip::Address::FreeAddr(gai);
 
-    // Perform NAT or TPROXY operations to retrieve the real client/dest IP addresses
-    if (conn->flags&(COMM_TRANSPARENT|COMM_INTERCEPTION) && !Ip::Interceptor.Lookup(details, conn)) {
-        debugs(50, DBG_IMPORTANT, "ERROR: NAT/TPROXY lookup failed to locate original IPs on " << details);
-        return Comm::NOMESSAGE;
+    if (conn->flags & COMM_TRANSPARENT) { // the real client/dest IP address must be already available via getsockname()
+        // XXX: a reconfiguration may 'stop' transparency in Interceptor (failing this check).
```

Comment:
IMO, This XXX case is worth adding error handling matching what happens in the NAT pathway instead of an assertion crash.

Reply:
My opinion is that if TransparentActive() is false we should not get here at all  - this means that such ports should not be marked with COMM_TRANSPARENT after reconfiguration.
NAT pathway is different because LookupNat() result depends on several factors, whereas TransparentActive() returns merely a flag value calculated at (re)configuration time.

Reply:
The situation is that (currently) new PortCfg objects are created on re-parsing configuration. The port pointed to by TcpAcceptor are the old ones that existed at the time the TcpAcceptor was initialized.

The only way for this XXX to be true/valid is when the Ip::Interceptor turns off transparency (new config), but the TcpAcceptor was created with an old port object (old config) and remained alive across the reconfigure. I do not recall without checking whether that can actually happen.

 * If you have checked and TcpAcceptor are stopped on reconfigure - then this XXX is wrong and it should just go away.
 * If TcpAcceptor remains active across reconfigure the XXX is valid and my request for handling the error stands.

Reply:
> Follow your suggestion

Done at 8e44db1.